sm3_asm.asm

```diff
@@ -88,13 +88,13 @@ sm3_compress_avx1 PROC
         mov	edx, DWORD PTR [ptr_L_SM3_AVX1_t]
         mov	ecx, r8d
         add	edx, r12d
-        roll	ecx, 12
+        rol	ecx, 12
```

Comment:
GCC complains that it needs the l to indicate the size the operand.
Have you tried compiling with GCC?
Assembly code generation may need to be fixed to handle this.

Reply:
I can't even figure out where the `rol` is coming from.  All of the `.rb files in wolfsm and ../scripts use `roll` or `rolq`, except a couple instances in `scripts/asm/x86_64/msvc_asm.rb`, which sounds out-of-scope.

Reply:
MSVC code here not for gcc.
This is correct.